analyzers/src/SonarAnalyzer.CSharp/Rules/CommentedOutCodeCodeFix.cs

```diff
@@ -0,0 +1,103 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Trivias = System.Collections.Generic.Dictionary<Microsoft.CodeAnalysis.FileLinePositionSpan, Microsoft.CodeAnalysis.SyntaxTrivia>;
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[ExportCodeFixProvider(LanguageNames.CSharp)]
+public sealed class CommentedOutCodeCodeFix : SonarCodeFix
+{
+    public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CommentedOutCode.DiagnosticId);
+
+    public override FixAllProvider GetFixAllProvider() => null;
+
+    protected override Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixContext context)
+    {
+        var diagnostic = context.Diagnostics.First();
+
+        context.RegisterCodeFix(CodeAction.Create(
+            CommentedOutCode.MessageFormat,
+            c =>
+            {
+                var comment = diagnostic.Location.GetLineSpan();
+                var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+
+                var newNode = node;
+
+                if (Remove(node.GetTrailingTrivia(), comment) is { } trailing)
+                {
+                    var token = node.ChildTokens().Last();
+                    ClearEmptiedLines(trailing, token, comment);
+                    RemoveWhitespace(trailing, token);
+                    newNode = node.WithTrailingTrivia(trailing.Values);
+                }
+                else if (Remove(node.GetLeadingTrivia(), comment) is { } leading)
+                {
+                    ClearEmptiedLines(leading, node.ChildTokens().First(), comment);
+                    newNode = node.WithLeadingTrivia(leading.Values);
+                }
+                else
```

Comment:
Why do you have this else block?

Reply:
Well, the good question is, what to do when you get there, because that can happen. So you could argue it is currently there for debug purposes.

Reply:
I think you should refactor this so that there is no else block. 
Either return inside the if and else-if branches, or always return the same thing, with `node==newNode` on the "useless" branch, so that the tree is the same.

Reply:
@gregory-paidis-sonarsource 

I've totally re-implemented the code fix.

It is now based on the token, instead of the node, as it turns out that not all trivia ar bound (leading or trailing) to any node.

Furthermore, I've decided that I'll only remove multi line comments if all lines are code (or whitespace, or `/*` or `*/`).